### DISCUSSION TITLE
websocket返回200

### DISCUSSION THREAD

**Discussion by hexergogo** [bfenetworks/bfe#889] [Q&A (问题和回答)] [answered]
![image](https://user-images.githubusercontent.com/46989742/141404063-91c67dec-9f1e-409e-9944-02157d098fd3.png)
配置了一个服务，里面有个js会启动一个websocket
![image](https://user-images.githubusercontent.com/46989742/141404117-57dba873-c4e9-49ed-8f5f-c1a31ace9247.png)

但是此websocket在处理后会返回200而不是101
![image](https://user-images.githubusercontent.com/46989742/141404166-cfd9536b-99fe-4fb0-a6cd-68c51b2db3db.png)

![image](https://user-images.githubusercontent.com/46989742/141404279-4a8470d1-1016-4ec6-858e-1c9a17aa04f1.png)

**Comment by hexergogo** [comment]
我试了试nginx就没有问题，不知道是为啥，求助

**Comment by daimg** [comment, marked as answer]
从描述来看，后端172.20.49.45直接返回了200，而不是101，所以js直接收到了200。
可以再查一下，为什么后端没有返回101。是否后端对BFE转发的消息的处理，产生了什么异常。

**Reply by hexergogo** [reply]
但是nginx做代理 就可以 bfe和nginx对websocket的处理有什么很大的区别么

**Reply by daimg** [reply]
从js发出的请求消息来看，似乎缺少websocket需要的header： Upgrade: websocket。你是否在nginx上为请求增加了这个header？

**Reply by hexergogo** [reply]
request里面是都有的 要不bfe也不会判断这是一个websocket 
![image](https://user-images.githubusercontent.com/46989742/141717549-223af010-118f-4e0a-ba49-69b89f73b344.png)

**Reply by hexergogo** [reply]
最后原因是因为 rewrite模块在websocket中不生效 改了改源码复用一下改写模块 可以用了 谢谢bfe
